ledger/cryptarchia-ledger/src/config.rs

```diff
@@ -31,15 +31,53 @@ impl Config {
     pub fn nonce_snapshot(&self, epoch: Epoch) -> Slot {
         let offset = self.base_period_length()
             * (self.epoch_period_nonce_buffer + self.epoch_stake_distribution_stabilization) as u64;
-        let base = u32::from(epoch) as u64 * self.epoch_length();
+        let base = (u32::from(epoch) - 1) as u64 * self.epoch_length();
```

Comment:
Is the reason we need `-1` that the nonce snapshot should be taken `7k/f` slots into the "previous" epoch?
(if `epoch_period_nonce_buffer + epoch_stake_distribution_stabilization` is 7)

Reply:
yes, the state for epoch `e` is taken at epoch `e-1` (not just for the nonce but stake distribution as well). It can't be the one in the current epoch `e` because if I take the snapshot of the nonce at offset `s` in epoch `e`, what do I use for slots earlier than `s` (but still in epoch `e`)?